syntax = "proto3";

package submission;
option go_package = "github.com/PowerLoom/proto-snapshot-collector/pkgs";

// Request structure as defined in your Solidity contract
message Request {
  uint64 deadline = 1;
  string snapshotCid = 2;
  uint64 epochId = 3;
  string projectId = 4;
}

message SnapshotSubmission {
  Request request = 1;
  bytes signature = 2;
  bytes signerPubKey = 3; // Public key in bytes
}

message ByteSubmission {
  bytes submission = 1;
}

service Submission {
  rpc SubmitSnapshot (stream ByteSubmission) returns (SubmissionResponse);
}

message SubmissionResponse {
  string message = 1; // Response message
}

